assembl/views/__init__.py

```diff
@@ -362,9 +363,104 @@ def process_export_list(ls):
         providers=providers,
         providers_json=json.dumps(providers),
         translations=io.open(jedfilename, encoding='utf-8').read(),
-        admin_email=admin_email
+        admin_email=admin_email,
     )
 
+    base.update({
+        "opengraph_locale": get_opengraph_locale(request),
+        "get_description": get_description(request),
+        "get_landing_page_image": get_landing_page_image(),
+        "private_social_sharing": private_social_sharing(),
+        "get_topic": get_topic(request)
+    })
+
+    return base
+
+
+def private_social_sharing():
+    """Returns true if the preference private_social_sharing is enabled. False otherwise"""
+    from ..auth.util import get_current_discussion
+    discussion = get_current_discussion()
+    if discussion:
+        return discussion.preferences["private_social_sharing"]
+    else:
+        return False
+
+
+def get_opengraph_locale(request):
+    """
+    If there is a user logged in, returns his preferred locale
+    If not, returns the first preferred locale of the discussion
+    Otherwise, sets locale to fr
+    """
+    from ..auth.util import get_user, get_current_discussion
+    from assembl.lib.locale import strip_country
+    user = get_user(request)
+    discussion = get_current_discussion()
+    if not user and not discussion:
+        locale = "fr"
+    elif user:
+        try:
+            locale = user.language_preference[0].locale.code
+            locale = strip_country(locale)
+        except:
+            if discussion:
+                locale = discussion.preferences['preferred_locales'][0]
+            else:
+                locale = "fr"
+    elif discussion and user is None:
+        locale = discussion.preferences['preferred_locales'][0]
+    return locale
+
+
+def get_description(request):
+    """
+    Returns the description corresponding to the locale returned from get_opengraph_locale
+    If the discussion does not have a description corresponding to this locale,
+    returns the description corresponding to the first preferred locale of the discussion
+    """
+    opengraph_locale = get_opengraph_locale(request)
+    from ..auth.util import get_current_discussion
+    discussion = get_current_discussion()
+    if discussion:
+        dict = discussion.preferences["extra_json"]
+        objectives_dict = dict.get("objectives", "default objectives")
+        if type(objectives_dict) == str:
+            return objectives_dict
+        else:
+            objectives_dict = objectives_dict["descriptionEntries"]
+            locale = discussion.preferences['preferred_locales'][0]
+            return objectives_dict.get(opengraph_locale, objectives_dict[locale])
```

Comment:
Did you make sure that whatever the content, a preview shows nicely? Namely to escape characters that could break <meta> tag structure or that could show not right in the social network's preview? Like quotes and html tags for example